### ISSUE TITLE
Parentheses around numeric literal followed by "." should not be recognized as gratuitous.

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#6864]
In the statement:

```js
var str = (1).toString();
```

The parentheses around 1 produce the error:

    Gratuitous parentheses around expression  no-extra-parens

But removing the parentheses results in a syntax error.

On the other hand, rewriting the statement as

```js
var str = 1..toString();
```

Results in

    A trailing decimal point can be confused with a dot  no-floating-decimal

It would be nice if ESLint allowed at least the first syntax, although both look fine to me.

**Comment by eslintbot** [comment]
Hi @fasttime, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by fasttime** [comment]
ESLint v3.2.2

**Comment by platinumazure** [comment]
For the decimal case, I think that rule can be suppressed using `1.0.toString()`, though that's not ideal.

I think I agree that this might be a bug in `no-extra-parens`, but let's see what the rest of the team thinks.

**Comment by vitorbal** [comment]
I was unable to reproduce this with the following config file and test code:

```json
{
    "root": true,
    "rules": {
        "no-extra-parens": "error"
    }
}
```

```javascript
var str = (1).toString();
```

@fasttime could you please post the ESLint configuration you are using that produces the error message mentioned?

**Comment by vitorbal** [comment]
@fasttime friendly ping, if you have a moment could you please provide updates on our questions above? Thanks!

**Comment by fasttime** [comment]
Sorry @vitorbal for the late reply. I can also no longer get the error reproduced. I haven't changed the ESLint configuration since then if that matters, just my own code. So it's possible that I just smoked something wrong when I posted this. Or some part of my old code was interfering with the parser and that's gone now. Not sure. Feel free to close this issue and sorry for the disturbance.

**Comment by vitorbal** [comment]
@fasttime no problem! Thanks for getting back to us. Glad the problem is fixed now :)
